### PR TITLE
Health Monitor - js errors on the https://www.fiverr.com/ site (close  #462)

### LINKED ISSUE #462: Health Monitor - js errors on the https://www.fiverr.com/ site

### PR DESCRIPTION
**Pull request by churkin**
\cc @AlexanderMoskovkin @miherlosev

### CODE REVIEW

**Comment by churkin** [comment]
@testcafe-build-bot \retest

**Comment by churkin** [comment]
FPR

File: src/client/sandbox/shadow-ui.js [hunk 1]
```diff
@@ -269,15 +271,37 @@ export default class ShadowUI extends SandboxBase {
         }
     }
 
+    overrideElement (el) {
+        if (domUtils.isBodyElement(el) || domUtils.isHeadElement(el)) {
```
Comment (miherlosev) [inline src/client/sandbox/shadow-ui.js]: We already overrode `head` and `body` elements on prototype level. see above.
`overrideElement` method should be removed.

File: src/client/sandbox/xhr.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            // NOTE: Emulate CORS, so that 3rd party libs (e.g. jQuery) allow requests with the proxy host as well as
+            // the destination page host.
+            if (!xhrSandbox.corsSupported)
+                this.withCredentials = false;
+
+            if (!this[NATIVE_BEHAVIOR]) {
+                var xhr = this;
+
+                try {
+                    url = getProxyUrl(url);
```
Comment (miherlosev) [inline src/client/sandbox/xhr.js]: remove `try, catch` statement because `getProxyUrl` don't raise error.

File: src/client/sandbox/xhr.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                }
+                catch (err) {
+                    xhrSandbox.emit(xhrSandbox.XHR_ERROR_EVENT, { err, xhr });
+
+                    return;
+                }
+            }
+
+            // NOTE: The 'async' argument is true by default. However, when the 'async' argument is set to â€˜undefinedâ€™,
```
Comment (miherlosev) [inline src/client/sandbox/xhr.js]: encoding problem for quotes

File: src/client/sandbox/xhr.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                xhrSandbox.emit(xhrSandbox.XHR_SEND_EVENT, { xhr });
+
+                var orscHandler = () => {
+                    if (this.readyState === 4)
+                        xhrSandbox.emit(xhrSandbox.XHR_COMPLETED_EVENT, { xhr });
+                };
+
+                // NOTE: If we're using the sync mode or the response is in cache and the object has been retrieved
+                // directly (IE6 & IE7), we need to raise the callback manually.
+                if (this.readyState === 4)
```
Comment (miherlosev) [inline src/client/sandbox/xhr.js]: IE6, IE7 is not supported browsers.
Do we need this statement now? If it is right then we should update the comment.

**Comment by miherlosev** [comment]
\r-

File: src/client/sandbox/xhr.js (hunk 4 above)
Comment (churkin) [inline src/client/sandbox/xhr.js]: This question should be studied separately

**Comment by churkin** [comment]
FPR

File: src/client/sandbox/xhr.js (hunk 4 above)
Comment (miherlosev) [inline src/client/sandbox/xhr.js]: I created issue https://github.com/DevExpress/testcafe-hammerhead/issues/489

**Comment by miherlosev** [comment]
lgtm

File: src/client/sandbox/xhr.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        this.XHR_ERROR_EVENT     = 'hammerhead|event|xhr-error';
+        this.XHR_SEND_EVENT      = 'hammerhead|event|xhr-send';
+
+        var xhr = new nativeMethods.XMLHttpRequest();
+
+        this.corsSupported = typeof xhr.withCredentials !== 'undefined';
+    }
+
+    static createNativeXHR () {
+        var xhr = new window.XMLHttpRequest();
```
Comment (AlexanderMoskovkin) [inline src/client/sandbox/xhr.js]: @churkin please clarify the following. The method is called `createNativeXHR`. We create `window.XMLHttpRequest` that can be overriden by some script. Should not we use `nativeMethods.XMLHttpRequest` here?

Comment (churkin) [inline src/client/sandbox/xhr.js]: Now we rewrite the `XMLHttpRequest` prototype. Therefore, actually, there are no more native `XMLHttpRequest `. In this case, we return an object that behaves like a native.

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**